src/pages/permissions/PermissionIdentities.tsx

```diff
@@ -0,0 +1,255 @@
+import {
+  Chip,
+  Row,
+  TablePagination,
+  useNotify,
+} from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { fetchIdentities } from "api/permissions";
+import Loader from "components/Loader";
+import ScrollableTable from "components/ScrollableTable";
+import SelectableMainTable from "components/SelectableMainTable";
+import SelectedTableNotification from "components/SelectedTableNotification";
+import { FC, useState } from "react";
+import { Link, useSearchParams } from "react-router-dom";
+import { queryKeys } from "util/queryKeys";
+import useSortTableData from "util/useSortTableData";
+import PermissionIdentitiesFilter, {
+  AUTH_METHOD,
+  PermissionIdentitiesFilterType,
+  QUERY,
+} from "./PermissionIdentitiesFilter";
+import { useSettings } from "context/useSettings";
+import PermissionIdentityEditGroupsBtn from "./PermissionIdentityEditGroupBtn";
+import { encodeIdentityNameForUrl } from "util/permissionIdentities";
+import usePanelParams, { panels } from "util/usePanelParams";
+import CustomLayout from "components/CustomLayout";
+import PageHeader from "components/PageHeader";
+import NotificationRow from "components/NotificationRow";
+import HelpLink from "components/HelpLink";
+import { useDocs } from "context/useDocs";
+import EditIdentityGroupsPanel from "./panels/EditIdentityGroupsPanel";
+
+const PermissionIdentities: FC = () => {
+  const notify = useNotify();
+  const {
+    data: identities = [],
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.identities],
+    queryFn: () => fetchIdentities(),
+  });
+  const { data: settings } = useSettings();
+  const docBaseLink = useDocs();
+  const panelParams = usePanelParams();
+  const [searchParams] = useSearchParams();
+  const [selectedIdentityNames, setSelectedIdentityNames] = useState<string[]>(
```

Comment:
selectedNames?

Reply:
I wanted to be a bit more specific with these variables because there are quite a few components with selectable tables and it gets confusing when you start navigating between files each with selectedNames, not sure if that makes sense?